src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -1278,4 +1240,204 @@ private bool ApplyVerbosityParameter(string parameterValue)
 
         #endregion
     }
+
+    /// <summary>
+    /// Console configuration needed for proper Console logging.
+    /// </summary>
+    internal interface IConsoleConfiguration
```

Comment:
one type per file

Reply:
'one type per file' is not followed in MSBuild code base. I had been asked in different PR to join it into one file.
We would have to negotiate new coding standards used for new MSBuild code.